src/components/ListTask/ListTask.jsx

```diff
@@ -1,28 +1,18 @@
+import styles from "./style.module.scss";
 import React from "react";
+import { useSelector, shallowEqual } from "react-redux";
 import Task from "../Task/Task";
-import PropTypes from "prop-types";
-import styles from "./style.module.scss";
 
-function ListTask(props) {
-  return (
-    <div className={styles.container}>
-      {props.todos.map((todo, index) => {
-        return (
-          <Task
-            todo={todo}
-            index={index}
-            key={todo.id}
-            onChange={props.onToggle}
-          ></Task>
-        );
-      })}
-    </div>
-  );
-}
+const selectTodoIds = (state) => state.todos.map((todo) => todo.id);
+
+const ListTask = () => {
+  const todoIds = useSelector(selectTodoIds, shallowEqual);
```

Comment:
why we use shallowEqual?

Reply:
If we toggle a todo item, the list of IDs will be considered the same, and <ListTask> won't have to re-render. The one <Task> will get an updated todo object and re-render, but all the rest of them will still have the existing todo object and not have to re-render at all.